internal/services/network/network_manager_resource.go

```diff
@@ -175,6 +176,23 @@ func (r ManagerResource) Create() sdk.ResourceFunc {
 				return metadata.ResourceRequiresImport(r.ResourceType(), id)
 			}
 
+			if len(state.Scope[0].ManagementGroups) > 0 {
+				metadata.Logger.Info("try to register 'Microsoft.Network' provider to the Management Group scope")
+				resourceProvidersClient := metadata.Client.Resource.ResourceProvidersClient
+				for _, group := range state.Scope[0].ManagementGroups {
+					groupId, err := managementGroupParse.ManagementGroupID(group)
+					if err != nil {
+						return err
+					}
+					_, err = resourceProvidersClient.RegisterAtManagementGroupScope(ctx, "Microsoft.Network", groupId.Name)
+					if err != nil {
+						return fmt.Errorf("registering the 'Microsoft.Network' provider with Management Group %s: %+v", groupId, err)
+					}
+				}
```

Comment:
can you give some more context on the issue here? Terraform shouldn't be silently registering resource providers at management group scopes which generally require extra-permissions